subsys/net/lib/zperf/zperf_tcp_receiver.c

```diff
@@ -32,26 +32,32 @@ static struct sockaddr_in *in4_addr_my;
 #define TCP_RECEIVER_THREAD_PRIORITY K_PRIO_PREEMPT(8)
 #endif
 
-#define TCP_RECEIVER_STACK_SIZE 2048
+#define TCP_RECEIVER_STACK_SIZE 4096
```

Comment:
Actually it's 5 threads by default, as we get one extra thread for the listener socket. After a brief initial review that struck me the most - this change increases memory requirement for zperf receiver a lot.

While I agree supporting several incoming connections is a nice feature, I wonder if we could handle this another way. Instead of spawning separate thread for each socket, we could have a single thread and monitor mutliple sockets with `poll()`. This function is capable of monitoring both, listener and receiver sockets at the same time so I don't see any technical constraints against such approach (see https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/net/sockets/echo_async).

Reply:
Would you agree on a KConfig to select if serving the connections by poll() or to serve them on separate threads? Separate threads would make sense when we have multiple cores available, while for single cpus it would make more sense using poll() and having less memory impact.
The poll() handling way could be selected by default by the KConfig

Reply:
> Would you agree on a KConfig to select if serving the connections by poll() or to serve them on separate threads?

I have a hard time justifying the need to maintain two separate implementations.

> Separate threads would make sense when we have multiple cores available

The majority of the packet processing takes place in a single thread anyway, so I really doubt it'd be the application layer that would be the bottleneck here.

Reply:
Hi @rlubos, yes I see your point, I will rework the last patch to handle multiple TCP connection by `poll()`, are you ok if I push this serie without the last patch? The other patches seems reasonable improvements.

Reply:
Sure, go ahead.

Reply:
Ack